src/test/java/org/opensearch/ad/rest/SecureADRestIT.java

```diff
@@ -1099,4 +1100,23 @@ public void testDeleteDetector() throws IOException {
 
     }
 
+    public void testInsightsApisUseSystemContextForJobIndex() throws IOException {
+        // Use a non-admin user with AD access (alice) to exercise Insights APIs end-to-end under security
+        String startPath = "/_plugins/_anomaly_detection/insights/_start";
+        String statusPath = "/_plugins/_anomaly_detection/insights/_status";
+        String stopPath = "/_plugins/_anomaly_detection/insights/_stop";
+
+        // Start insights job as alice
+        Response startResp = TestHelpers.makeRequest(aliceClient, "POST", startPath, ImmutableMap.of(), "", null);
+        assertEquals("Start insights job failed", RestStatus.OK, TestHelpers.restStatus(startResp));
+
+        // Status should be accessible and return OK for the same non-admin user
+        Response statusResp = TestHelpers.makeRequest(aliceClient, "GET", statusPath, ImmutableMap.of(), "", null);
+        assertEquals("Get insights job status failed", RestStatus.OK, TestHelpers.restStatus(statusResp));
+
+        // Stop should also be accessible and return OK
+        Response stopResp = TestHelpers.makeRequest(aliceClient, "POST", stopPath, ImmutableMap.of(), "", null);
+        assertEquals("Stop insights job failed", RestStatus.OK, TestHelpers.restStatus(stopResp));
+    }
+
```

Comment:
Can you have alice create detectors, check if results are generated, start insights, check if any insights can be generated after a few minutes; query insights using normal user; then stop insights?